src/Program.cs

```diff
@@ -1,245 +1,263 @@
 ﻿using System;
-using System.Net.Http;
-using System.Net.Http.Headers;
-using System.Text;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Configuration;
 using System.Threading;
-using Microsoft.IdentityModel.Clients.ActiveDirectory;
-using Newtonsoft.Json;
+
+using Microsoft.Azure.Management.Search;
+using Microsoft.Azure.Management.Search.Models;
+using Microsoft.Azure.Management.ResourceManager;
+using Microsoft.Azure.Management.ResourceManager.Models;
+using Sku = Microsoft.Azure.Management.Search.Models.Sku;
+
+using Microsoft.Rest.Azure.Authentication;
+
 
 namespace ManagementAPI
 {
-    // In order to use the Azure Resource Manager API you need to prepare the target subscription. This is discussed
-    // in detail here:
-    // http://msdn.microsoft.com/en-us/library/azure/dn790557.aspx
+    // The Azure Cognitive Search Management SDK authentication uses a service principal in Azure Active Directory.  Before you can run the sample, you'll need to
+    // create a service principal and give it the proper permissions in your subscription.  
+    // This is discussed in detail here: https://docs.microsoft.com/azure/active-directory/develop/howto-create-service-principal-portal
 
     class Program
     {
-        // You can obtain this information from the Azure management portal. The instructions in the link above 
-        // include details for this as well.
-        private const string TenantId = "<your tenant id>";
-        private const string ClientId = "<your client id>";
-        private const string SubscriptionId = "<your subscription id>";
-
-        // This is the return URL you configure during AD client application setup. For this type of apps (non-web apps)
-        // you can set this to something like http://localhost/testapp. The important thing is that the URL here and the
-        // URL in AD configuration match.
-        private static readonly Uri RedirectUrl = new Uri("<your redirect url");
+        public static void Main(string[] args)
+        {
+            IConfigurationBuilder builder = new ConfigurationBuilder().AddJsonFile("appsettings.json");
+            IConfigurationRoot configuration = builder.Build();
 
-        private static readonly Random _random = new Random();
+            // Updated this information in the appsettings.json in this project.
+            // See the instructions at the link above for the service principal details and the subscription id.
+            var tenantId = configuration["TenantId"];
+            var clientId = configuration["ClientId"];
+            var clientSecret = configuration["ClientSecret"];
+            var subscriptionId = configuration["SubscriptionId"];
 
-        private static string _authorizationToken = null;
+            if (new List<string> { tenantId, clientId, clientSecret, subscriptionId }.Any(i => String.IsNullOrEmpty(i)))
```

Comment:
nit: No need to create a list. This could just be:

```csharp
if (new[] { tenantId, clientId, clientSecret, subscriptionId }.Any(i => String.IsNullOrEmpty(i)))
{
…
}
```
```

Reply:
fixed